adcc/ExcitedStates.py

```diff
@@ -657,6 +659,66 @@ def describe_amplitudes(self, tolerance=0.01, index_format=None):
                 ret += separator
         return ret[:-1]
 
+    def to_dataframe(self):
+        """
+        Exports the ExcitedStates object as :class:`pandas.DataFrame`.
+        All values are printed in atomic units.
+        """
+        propkeys = self.excitation_property_keys.copy()
+        propkeys.extend(self.excitation_energy_corrections.keys())
+        data = {
+            "excitation": np.arange(0, self.size, dtype=int),
+            "kind": np.tile(self.kind, self.size)
+        }
+        # fill the dict with data, sort out stuff that we cannot represent
+        # in a table
+        unacceptable_types = [adcc.AmplitudeVector, adcc.OneParticleOperator,
+                              adcc.Tensor]
```

Comment:
In these kind of situations there is always the question of blacklisting versus whitelisting. Generally I think whitelisting is better here, because it is more forward-compatible (you don't want this to break if you implement a new fancy property ... better it just does not show up). So I'd vote for whitelisting numbers and np arrays and lists and the few other things (if any). If you use proper subtypes this is not too many tests (numpy has base types for all numbers I think).

Reply:
Probably we only need ndarrays... we don‘t really use lists for numeric types and I don’t think it should be encouraged :smile:
So I‘d say we only export numpy ndarrays which are subtypes of np.numeric. Currently I cannot imagine anything one might „compute“ and mark as excitation property that would be a string...

Reply:
I agree should be as specific as sensible. It's just the first few things which came to my mind, so I wrote them down, but just all subtypes of np.numeric sounds exactly what we need (just make sure this includes standard python ints and such).

Reply:
It's much simpler and easier to read now.

Reply:
I agree :+1: